### PR TITLE
feat(route): add watasuke.net

### PR DESCRIPTION
**Pull request by honahuku**
## Involved Issue / 该 PR 相关 Issue

No related issues

## Example for the Proposed Route(s) / 路由地址示例

```routes
/watasuke/blog
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
This watasuke.net is the site of @watasuke102, but I have received his permission to send this PR.

### CODE REVIEW

File: website/docs/routes/blog.mdx
Comment (honahuku) [inline website/docs/routes/blog.mdx:1]: I have one question.
I added it to blog.mdx and ran `npm run format` as per the documentation. Then, I was told that `pinyin-pro` does not exist, so I executed `npm i pinyin-pro` and `npm run format` was successful, but is this what you intended?

Comment (TonyRL) [reply website/docs/routes/blog.mdx:1]: > Run the following command in the `website` **directory**:
>
> ```
> pnpm i
>```

Comment (honahuku) [reply website/docs/routes/blog.mdx:1]: thank you! ! It worked

File: lib/routes/watasuke/blog.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        return {
+            title: article.title,
+            description: ogDescription,
+            pubDate: parseDate(article.published_at),
+            updated: parseDate(article.updated_at),
+            link: articleUrl,
+            category: article.tags.map((tag) => tag.name),
+            itunes_item_image: ogImage,
+        };
+    });
```
Comment (TonyRL) [inline lib/routes/watasuke/blog.js:32]: Please use cache if you retrieve the full article.

File: lib/routes/watasuke/blog.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+import cheerio from 'cheerio';
```
Comment (TonyRL) [inline lib/routes/watasuke/blog.js:3]: Use
```js
import * as cheerio from 'cheerio';
```
or 
```js
import { load } from 'cheerio';
```

File: lib/routes/watasuke/blog.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            link: articleUrl,
+            category: article.tags.map((tag) => tag.name),
+            itunes_item_image: ogImage,
+        };
+    });
+
+    const items = await Promise.all(itemsPromises);
+
+    ctx.set('data', {
+        title: ogTitle || String(baseUrl),
```
Comment (TonyRL) [inline lib/routes/watasuke/blog.js:37]: https://github.com/DIYgod/RSSHub/blob/9ea5161178385d289cf29458fe8f645fd8352460/lib/routes/watasuke/blog.js#L5

is already a string. No need to type cast it again.

File: lib/routes/watasuke/blog.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    const blogPageHtml = await got(`${baseUrl}/blog/`).text();
+    const $ = cheerio.load(blogPageHtml);
+    const ogTitle = $('meta[property="og:title"]').attr('content');
+
+    const apiUrl = `${baseUrl}/page-data/blog/page-data.json`;
+    const response = await got(apiUrl);
+    const articles = response.data.result.data.allArticles.nodes;
+
+    const itemsPromises = articles.map(async (article) => {
+        const articleUrl = `${baseUrl}/blog/article/${article.slug}`;
```
Comment (TonyRL) [inline lib/routes/watasuke/blog.js:17]: Please use the redirected url instead.
logs: https://github.com/DIYgod/RSSHub/actions/runs/8128222519/job/22213891885#step:12:56

File: lib/routes/watasuke/blog.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        const articleHtml = await got(articleUrl).text();
```
Comment (TonyRL) [inline lib/routes/watasuke/blog.js:18]: The full article body in Markdown is already provided from
https://github.com/DIYgod/RSSHub/blob/9ea5161178385d289cf29458fe8f645fd8352460/lib/routes/watasuke/blog.js#L12
No need to visit the site again for the same content.

File: lib/routes/watasuke/maintainer.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (TonyRL) [inline lib/routes/watasuke/maintainer.js:1]:
```suggestion
export default {
```

File: lib/routes/watasuke/radar.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
Comment (TonyRL) [inline lib/routes/watasuke/radar.js:1]:
```suggestion
export default {
```

**Pull request merged**